modules/heat_conduction/include/bcs/ConvectiveHeatFluxBC.h

```diff
@@ -0,0 +1,42 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "IntegratedBC.h"
+
+class ConvectiveHeatFluxBC;
+
+template <>
+InputParameters validParams<ConvectiveHeatFluxBC>();
+
+/**
+ * Boundary condition for convective heat flux where temperature and heat transfer coefficient are
+ * given by material properties.
+ */
+class ConvectiveHeatFluxBC : public IntegratedBC
+{
+public:
+  static InputParameters validParams();
+
+  ConvectiveHeatFluxBC(const InputParameters & parameters);
+
+protected:
+  virtual Real computeQpResidual() override;
+  virtual Real computeQpJacobian() override;
+
+  /// Far-field temperatue fields for each component
```

Comment:
Just "Far-field temperature field", i.e. (a) fix the typo and (b) you are dealing with just one field.